File: September/2026-09-04/409/409-longest-palindrome.py
# 409-longest-palindrome.cpp;help-syntax;map;leetcode

class Solution:
    def longestPalindrome(self, s: str) -> int:
        mp = {}

        for ch in s:
            mp[ch] = mp.get(ch, 0) + 1

        length = 0
        hasOdd = False

        for count in mp.values():
            length += (count // 2) * 2

            if count % 2 == 1:
                hasOdd = True

        if hasOdd:
            length += 1

        return length

File: September 2026/2026-09-04/409-longest-palindrome.cpp
// 409-longest-palindrome.cpp;help-syntax;map;leetcode

class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> mp;

        for(char ch : s) {
            mp[ch]++;
        }

        int length = 0;
        bool hasOdd = false;

        for(auto it : mp) {
            int count = it.second;

            length += (count / 2) * 2;

            if(count % 2 == 1) {
                hasOdd = true;
            }
        }

        if(hasOdd) {
            length++;
        }

        return length;
    }
};